src/jrd/os/win32/winnt.cpp

```diff
@@ -107,7 +107,7 @@ using namespace Firebird;
 
 static bool	maybeCloseFile(HANDLE&);
 static jrd_file* seek_file(jrd_file*, BufferDesc*, OVERLAPPED*);
-static jrd_file* setup_file(Database*, const Firebird::PathName&, HANDLE, bool, bool);
+static jrd_file* setup_file(Database*, const Firebird::PathName&, HANDLE, bool, bool, bool, bool);
```

Comment:
Having 4 bool parameters in function makes code bad readable. Even 2 were not too good, but 4 is awful. Suggest to replace with set of flags like PIO_FORCE_WRITE, PIO_USE_FILESYSTEM_CACHE, etc and single flags parameter. 

Also please pay attention to avoiding flags like not_USE_FILESYSTEM_CACHE, flag named USE_FILESYSTEM_CACHE instead is much better readable.

Reply:
We already have corresponding `FIL_*` flags. But this change would mean that we need to convert from multiple bools to flags at least twice inside the code, before every `setup_file` call. So it would complicate the code, thus I avoided that. But I will reconsider.

As for `notSomething`, generally I agree with you, but this particular name was invented before me and I just preserved status quo. And it was added to handle a special and relatively rare case, so optional "NO" flag looked slightly better to me than [almost] mandatory "YES" flag. But I can change this too, if required. Any other opinions?